pywbem/_listener.py

```diff
@@ -893,6 +905,9 @@ def add_server(self, server):
         else:
             raise ValueError("Invalid scheme in server URL: %s" % \
                              server.conn.url)
+                             
+        # set the prefix into the server object
+        server.listener_id = self._listener_id
```

Comment:
I don't like that at all; it creates a new instance variable from outside of the class, that has no real semantic meaning for that class, and that uses the object just as a store to pass it on to when it is needed.

Alternative proposal, removing the need for adding the `listener_id` instance variable to `WBEMServer`:
* For `_create_destination()`, add a `listener_id` parameter, and when calling it in `add_server()`, pass `self.__listener_id` as its value.
* For `_create_filter()`, add a `listener_id` parameter, and when calling it in `add_dynamic_filter()`, pass `self.__listener_id` as its value.